SU-HSDO/suhumsci
# STN-471: Banner

# READY FOR REVIEW

## Summary
This pull request adds a new paragraph type called Banner that is available in the components section and the hero region. The only content allowed in the banner area is either one image or one video and the media used scale to fit the entire width of the hero area.

**Banner in hero with image:**
![Banner___QA_Site__Colorful_on_Stage__Traditional_on_Dev_](https://user-images.githubusercontent.com/2486846/89554151-768dc180-d7dc-11ea-9cec-f0f12c8e44b9.png)

**Banner in hero with video:**
![Banner_w__Video___QA_Site__Colorful_on_Stage__Traditional_on_Dev_](https://user-images.githubusercontent.com/2486846/89554201-89a09180-d7dc-11ea-9d1f-38b9c4cfef36.png)

**Banners in content area including row wells:**
![Banner___QA_Site__Colorful_on_Stage__Traditional_on_Dev_](https://user-images.githubusercontent.com/2486846/89554362-c2406b00-d7dc-11ea-945f-ecaf3b5dcd6e.png)

This work has been tested in the following browsers/devices:
- [x] iPhone 11 (Safari)

## Need Review By (Date)
08/10

## Urgency
medium

## Steps to Test
1. Ensure that all tests are passing
2. Ensure that new behat tests are passing `lando behat --tags='@install&&~@javascript'` (some existing tests do not pass when ran locally)
3. Create a new page using the banner component in the hero region. Test using both an image and then a video. 
    - [x] The image should use the `Hero Image` image style
    - [x] Videos should span the entire width of the page and maintain a 16:9 aspect ratio (it will be huge)
4. Add banners with images and videos to the components section
    - [x] Images should use the `Hero Image` image style and fill only the component area
    - [x] Videos should span the entire width of the component area and maintain a 16:9 aspect ratio
5. Add banners with images and videos to rows using the limited and full width options.
    - [x] Images and videos behave as they do in the component section outside of the rows except with appropriate spacing within the wells.
6. Test that both themes behave the same and maintain proper spacing
    -  [x] Colorful
    -  [x] Traditional

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
